tests/units/TicketTest.php

```diff
@@ -294,4 +294,81 @@ public function testEscalationWithMandatoryFields()
         }
         $this->assertTrue($found_new_group, "New group should be assigned after successful escalation");
     }
+
+    public function testEscalateButtonShouldTriggerGroupEscalationAndExecuteRuleOnTicket()
+    {
+        $this->login();
+
+        $config = new PluginEscaladeConfig();
+        $conf = $config->find();
+        $conf = reset($conf);
+        $config->getFromDB($conf['id']);
+        $this->assertGreaterThan(0, $conf['id']);
+
+        PluginEscaladeConfig::loadInSession();
+
+        $group_observer = new \Group();
+        $group_observer_id = $group_observer->add([
+            'name' => 'Group Observer',
+            'entities_id' => 0,
+            'is_recursive' => 1,
+        ]);
+        $this->assertGreaterThan(0, $group_observer_id);
```

Comment:
Please use `createItem()`, it's more efficient.
```suggestion
        $group_observer_id = $this->createItem(\Group::class, [
            'name' => 'Group Observer',
            'entities_id' => 0,
            'is_recursive' => 1,
        ])->getID();
```

Do it everywhere after